From b27a06aa7b7160c5734c36756dbd72348e7af3f6 Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Thu, 4 Jun 2015 16:56:49 +0800
Subject: [PATCH 416/462] ov2722: power up sequence update

DOVDD(1,8V) should be up before AVDD(2,8V),
according to specifications.

Change-Id: I1879c4d2268bae7b472029ba9b39c9c2a287034d
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-37623
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/58886
Build: IMIN (SC)
Reviewed-by: Bingbu Cao <bingbu.cao@intel.com>
Approver: Bingbu Cao <bingbu.cao@intel.com>
Reviewed-by: Jian Xu Zheng <jian.xu.zheng@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov2722.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
index 598dbcd..1dfee37 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
@@ -724,11 +724,11 @@ static int power_ctrl(struct v4l2_subdev *sd, bool flag)
 
 #ifdef CONFIG_GMIN_INTEL_MID
 	if (flag) {
-		ret = dev->platform_data->v2p8_ctrl(sd, 1);
+		ret = dev->platform_data->v1p8_ctrl(sd, 1);
 		if (ret == 0) {
-			ret = dev->platform_data->v1p8_ctrl(sd, 1);
+			ret = dev->platform_data->v2p8_ctrl(sd, 1);
 			if (ret)
-				dev->platform_data->v2p8_ctrl(sd, 0);
+				dev->platform_data->v1p8_ctrl(sd, 0);
 		}
 	} else {
 		ret = dev->platform_data->v1p8_ctrl(sd, 0);
-- 
1.9.1

